File: hr_ge_parser/app/schemas/company.py
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel, ConfigDict


class CompanyBase(BaseModel):
    name: str
    name_en: Optional[str] = None
    logo_url: Optional[str] = None
    thumbnail_url: Optional[str] = None
    is_anonymous: Optional[bool] = False


class CompanyResponse(CompanyBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    external_id: int
    cover_image_url: Optional[str] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
    job_count: Optional[int] = None


class PaginationMeta(BaseModel):
    total: int
    page: int
    per_page: int
    total_pages: int


class CompanyListResponse(BaseModel):
    data: List[CompanyResponse]
    meta: PaginationMeta

File: hr_ge_parser/frontend/src/types/company.ts
import type { PaginationMeta } from './common';

// ============================================================
// BASE TYPES
// ============================================================

export interface Company {
  id: number;
  external_id: number;
  name: string;
  name_en: string | null;
  logo_url: string | null;
  thumbnail_url: string | null;
  cover_image_url: string | null;
  is_anonymous: boolean;
  is_blacklisted: boolean;
  status_id: number;
  job_count?: number;
  created_at: string | null;
  updated_at: string | null;
}

// ============================================================
// API TYPES
// ============================================================

export interface CompanyFilters {
  page?: number;
  per_page?: number;
  search?: string;
  is_anonymous?: boolean;
  sort_by?: 'name' | 'job_count' | 'created_at';
  sort_order?: 'asc' | 'desc';
}

export interface CompaniesResponse {
  data: Company[];
  meta: PaginationMeta;
}
